packages/traceloop-sdk/traceloop/sdk/evaluator/stream_client.py

```
@@ -0,0 +1,91 @@
+
+            if parsed_result.execution_id != execution_id:
+                raise Exception(
+                    f"Execution ID mismatch: {parsed_result.execution_id} != {execution_id}"
+                )
+
+            return parsed_result
+
+        except httpx.ConnectError as e:
+            raise Exception(
+                f"Failed to connect to stream URL: {full_stream_url}. Error: {e}"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)